pandas/_libs/tslibs/offsets.pyx

```diff
@@ -212,13 +212,66 @@ def _validate_business_time(t_input):
             raise ValueError("time data must match '%H:%M' format")
     elif isinstance(t_input, dt_time):
         if t_input.second != 0 or t_input.microsecond != 0:
-            raise ValueError(
-                "time data must be specified only with hour and minute")
+            raise ValueError("time data must be specified only with "
+                             "hour and minute")
         return t_input
     else:
         raise ValueError("time data must be string or datetime.time")
 
 
+def _validate_n(n):
+    """
+    Require that `n` be a nonzero integer.
+
+    Parameters
+    ----------
+    n : int
+
+    Raises
+    ------
+    ValueError
+    """
+    if n == 0 or not isinstance(n, int):
+        raise ValueError("N cannot be 0")
+
+
+def _validate_month(month):
+    """
+    Require that `month` be an integer between 1 and 12 inclusive.
+
+    Parameters
+    ----------
+    month : int
+
+    Raises
+    ------
+    ValueError
+    """
+    if not isinstance(month, int) or not 1 <= month <= 12:
+        raise ValueError("Month must go from 1 to 12")
+
+
+def _validate_weekday(weekday, allow_none=False):
+    """
```

Comment:
if this is a method, then you don't need allow_none (which is a really odd non-standard kw).

because you are a method, checking ``self.weekday`` is obvious.

now you *can* also have a _validate_weekday method that does everything (but the allow_none) and simply call that if you want.

If you show that you need these in multiple places, then certainly you can call them, from the offsets you need to have a method.

Reply:
I still don't totally understand the reasoning, but it sounds like the upshot here is that to get this done `_validate_weekday` needs to become a method of `_BaseOffset`, and without a `allow_none` kwarg.  Is that the correct takeaway?